tests/unit/processor/dissecter/test_dissecter.py

```diff
@@ -0,0 +1,473 @@
+        {"message": "This is the message", "extracted": ["message", "the", "is", "This"]},
+    ),
+    (
+        "append to existing prefilled field in different order as list",
+        {
+            "filter": "message",
+            "dissecter": {
+                "mapping": {
+                    "message": "%{+extracted/4} %{+extracted/3} %{+extracted/2} %{+extracted/1}"
+                }
+            },
+        },
+        {"message": "This is the message", "extracted": ["preexisting"]},
+        {
+            "message": "This is the message",
+            "extracted": ["preexisting", "message", "the", "is", "This"],
+        },
+    ),
+    (
+        "append to new field in specified order as string with multiple fields",
+        {
+            "filter": "message",
+            "dissecter": {
+                "mapping": {
+                    "message": "%{}: %{+extracted/2}",
+                    "message2": "%{}: %{+extracted/1}",
+                }
+            },
+        },
+        {"message": "The first message: first", "message2": "The second message: second"},
+        {
+            "message": "The first message: first",
+            "message2": "The second message: second",
+            "extracted": "second first",
+        },
+    ),
+    (
+        "converts datatype without mapping",
+        {"filter": "message", "dissecter": {"convert_datatype": {"message": "int"}}},
+        {"message": "42"},
+        {"message": 42},
+    ),
+    (
+        "converts datatype with mapping in dotted field notation",
+        {
+            "filter": "message",
+            "dissecter": {
+                "mapping": {
+                    "message": "%{}of %{extracted.message_float} and a int of %{extracted.message_int}"  # pylint: disable=line-too-long
+                },
+                "convert_datatype": {
+                    "extracted.message_int": "int",
+                    "extracted.message_float": "float",
+                },
+            },
+        },
+        {"message": "This message has a float of 1.23 and a int of 1337"},
+        {
+            "message": "This message has a float of 1.23 and a int of 1337",
+            "extracted": {"message_float": 1.23, "message_int": 1337},
+        },
+    ),
+    (
+        "indirect field notation: uses captured field as key",
+        {"filter": "message", "dissecter": {"mapping": {"message": "%{?key} %{&key}"}}},
+        {"message": "This is the message"},
+        {"message": "This is the message", "This": "is the message"},
+    ),
+    (
+        "indirect field notation: uses captured field as key and append to it",
+        {
+            "filter": "message",
+            "dissecter": {"mapping": {"message": "%{?key} %{&key} %{} %{+&key}"}},
+        },
+        {"message": "This is the message"},
+        {"message": "This is the message", "This": "is message"},
+    ),
+]
+failure_test_cases = [  # testcase, rule, event, expected
```

Comment:
I really like the failure tags, but if you want to spend a bit more time on it then I think it would be great if they would have more content to them. Like a short indication why the dissect failed. Like `_dissectfailure_not_supported_type` or `_dissectfailure_mapping_field_not_found`. Maybe we should discuss again though, how long these tags should be.

Reply:
it is upon the rule write to declare what is written in this tag. I wanted to stay close to the original logstash behavior.

Reply:
well if I understand it correctly the rule write can't implement some form of logic that says when to use which tag. So my idea was to add some default failure tags. But it is also not a fixed requirement for now. We can still extend it later when the need arises.